### PR TITLE
[SYCL] Fix free function queries for host device

### PR DESCRIPTION
**Pull request by alexbatashev**
https://github.com/intel/llvm/pull/4090 moved free function queries to a different namespace. Host device implementation actually relies on integration header to extract information about free function queries and set the appropriate values.

### CODE REVIEW

**Review by AaronBallman — DISMISSED** [review]
LGTM, but a better commit message wouldn't make me sad.

File: clang/test/CodeGenSYCL/parallel_for_this_item.cpp [hunk 1]
```diff
@@ -114,13 +166,31 @@
...
+// CHECK-NEXT:};
 
 #include "sycl.hpp"
 
 using namespace cl::sycl;
 
 SYCL_EXTERNAL item<1> g() { return this_item<1>(); }
 SYCL_EXTERNAL item<1> f() { return g(); }
+SYCL_EXTERNAL item<1> s() { return ext::oneapi::experimental::this_item<1>(); }
+SYCL_EXTERNAL item<1> h() { return s(); }
```
Comment (elizabethandrews) [inline clang/test/CodeGenSYCL/parallel_for_this_item.cpp:193]: What is the purpose of this? I don't see a test with this.

Comment (alexbatashev) [reply clang/test/CodeGenSYCL/parallel_for_this_item.cpp:193]: Good catch! Thanks! I updated the test to call `h()` instead of `s()`.

**Review by elizabethandrews — APPROVED** [review]
LGTM. Thanks!

**Comment by alexbatashev** [comment]
The failing test seems to be unrelated to this PR and affects other PRs as well. I created a tracker and a PR to disable this test: https://github.com/intel/llvm/pull/4368

**Pull request merged**